tests/test-extensions/petstore-extension/src/main/java/org/mule/test/petstore/extension/PetStoreConnector.java

```diff
@@ -28,9 +28,9 @@
     PooledPetStoreConnectionProviderWithValidConnection.class})
 @ErrorTypes(PetstoreErrorTypeDefinition.class)
 @Sources({PetStoreSource.class, FailingPetStoreSource.class, SentientSource.class, PetAdoptionSource.class,
-    PetStoreStreamSource.class, PetStoreListSource.class, PartialPetAdoptionSource.class, NumberPetAdoptionSource.class,
-    ConnectedPetAdoptionSource.class, PetFailingPollingSource.class, WatermarkingPetAdoptionSource.class,
-    PetAdoptionLimitingSource.class})
+    PetStoreStreamSource.class, PartialPetAdoptionSource.class, ConnectedPetAdoptionSource.class, PetFailingPollingSource.class,})
+@org.mule.sdk.api.annotation.Sources({NumberPetAdoptionSource.class, WatermarkingPetAdoptionSource.class,
```

Comment:
this is taking an "all or nothing" approach, meaning that you either have sources fully defined with the legacy API or fully defined with the new API. Add cases for sources that mix both.